File: ios/HydrateAI/SharedHydrationModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SharedHydrationModule, NSObject)
RCT_EXTERN_METHOD(write:(nonnull NSNumber *)consumedOz
                  goalOz:(nonnull NSNumber *)goalOz
       nextDrinkMinutes:(nullable NSNumber *)nextDrinkMinutes)
+ (BOOL)requiresMainQueueSetup { return YES; }
@end 

File: ios/HydrateAI/SharedHydrationModule.swift
import Foundation
import React

@objc(SharedHydrationModule)
class SharedHydrationModule: NSObject, RCTBridgeModule {
  static func moduleName() -> String! { "SharedHydrationModule" }
  static func requiresMainQueueSetup() -> Bool { true }

  @objc
  func write(_ consumedOz: NSNumber, goalOz: NSNumber, nextDrinkMinutes: NSNumber?) {
    SharedHydrationStore.write(consumedOz: consumedOz, goalOz: goalOz, nextDrinkMinutes: nextDrinkMinutes)
  }
} 